### PR TITLE
SITL: enable ROMFS & metadata storage

### PR DESCRIPTION
**Pull request by bkueng**
This enables the ROMFS for SITL, the same way as for NuttX builds, just w/o the image generation.
That allows to include generated files, such as the parameters metadata (here enabled for SITL for testing).

ROMFS files are now copied to the `build_<board>/etc` directory, requiring a change of SITL/posix startup parameters.

@DonLakeFlyer  @hamishwillee in SITL you can now use mavftp to request the parameter metadata under `/etc/extras/parameters.json.gz`.

Not tested with ROS yet.

### CODE REVIEW

**Comment by DonLakeFlyer** [comment]
> in SITL you can now use mavftp to request the parameter metadata under `/etc/extras/parameters.json.gz`

A step in the right direction. But still need the rest of the `COMPONENT_INFORMATION` protocol to tell QGC where this is.

File: ROMFS/px4fmu_common/CMakeLists.txt [hunk 1]
```diff
@@ -33,3 +33,8 @@
 
 add_subdirectory(init.d)
 add_subdirectory(mixers)
+# TODO: make this configurable from the board config
```
Comment (dagar) [inline ROMFS/px4fmu_common/CMakeLists.txt:36]: I'd rather try to merge them entirely.

File: boards/px4/sitl/default.cmake [hunk 2]
```diff
@@ -5,6 +5,7 @@ px4_add_board(
 	MODEL sitl
 	ROMFSROOT px4fmu_common
 	LABEL default
+	EMBEDDED_METADATA parameters
```
Comment (dagar) [inline boards/px4/sitl/default.cmake:8]: What if we just did this by default everywhere unless `CONSTRAINED_FLASH`?

File: ROMFS/px4fmu_common/CMakeLists.txt (hunk 1 above)
Comment (bkueng) [reply ROMFS/px4fmu_common/CMakeLists.txt:36]: Yeah even better

File: boards/px4/sitl/default.cmake (hunk 2 above)
Comment (bkueng) [reply boards/px4/sitl/default.cmake:8]: It's a possibility, I'm not sure yet how the end result should look like. For now I wanted something to selectively enable for testing w/o affecting other targets.
I can imagine that some targets might want to use external metadata (e.g. from a companion) for faster download.

File: ROMFS/CMakeLists.txt [hunk 3]
```diff
@@ -179,17 +184,35 @@ foreach(board_rc_file ${OPTIONAL_BOARD_RC})
...
+if (config_romfs_extra_files)
+	set(extras_copy_cmd COMMAND ${CMAKE_COMMAND} -E copy_if_different ${config_romfs_extra_files} ${romfs_gen_root_dir}/extras/)
+else()
+	set(extras_copy_cmd "")
+endif()
 add_custom_command(OUTPUT romfs_extras.stamp
 	COMMAND ${CMAKE_COMMAND} -E make_directory ${romfs_gen_root_dir}/extras/
 	COMMAND ${CMAKE_COMMAND} -E make_directory ${PX4_BINARY_DIR}/romfs_extras/
 	COMMAND ${CMAKE_COMMAND} -E copy_directory ${PX4_BINARY_DIR}/romfs_extras/ ${romfs_gen_root_dir}/extras/
+	${extras_copy_cmd}
```
Comment (dagar) [inline ROMFS/CMakeLists.txt:200]: TIL. CMake continues to surprise me (not in a good way).

Side note, the dependencies here were never quite right. I believe we'll ultimately need to have cmake more directly handle the pruning to do it perfectly.

**Review by dagar — APPROVED** [review]

**Pull request merged**